cumulusci/tasks/metadata_etl/layouts.py

```diff
@@ -68,6 +69,262 @@ def _create_new_related_list(self, metadata, api_name, related_list):
+                - read_only: Boolean (default False, not compatable with required)
+                - section: [index] (Use with relative: top, bottom)
+                - column: [first | last] (Use with relative: top, bottom)
+        """
+    entity = "Layout"
+    task_options = {
+        "fields": {
+            "description": "List of fields. See task info for structure.",
+            "required": False,
+        },
+        "pages": {
+            "description": "List of Visualforce Pages. See task info for structure.",
+            "required": False,
+        },
+        **MetadataSingleEntityTransformTask.task_options,
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        self._adding_fields = process_list_arg(self.options.get("fields", []))
+        self._adding_pages = process_list_arg(self.options.get("pages", []))
+        self._existing_names = []
+
+    def _transform_entity(
+        self, metadata: MetadataElement, api_name: str
+    ) -> Optional[MetadataElement]:
+
+        self._metadata = metadata
+
+        self._collect_existing_items()
+
+        self._set_adding_fields(api_name)
+        self._set_adding_pages(api_name)
+
+        return metadata
+
+    ## Collect fields already in the layout
+    def _collect_existing_items(self):
+        ## Need to traverse all layoutSections->layoutColumns->layoutItems->fields to find if item/field already exists
+        layout_sections = self._metadata.findall("layoutSections")
+        for section in layout_sections:
+            layout_columns = section.findall("layoutColumns")
+            for col in layout_columns:
+                layout_items = col.findall("layoutItems")
+                for item in layout_items:
+                    if item.find("page"):
+                        self._existing_names.append(item.page.text)
+                    elif item.find("field"):
+                        self._existing_names.append(item.field.text)
+
+    def _set_adding_fields(self, api_name):
+        # Do nothing if there are no fields
+        if len(self._adding_fields) == 0:
+            return
+
+        field_props = {field["api_name"]: field for field in self._adding_fields}
+
+        layout_item_dict = self._add_items(self._adding_fields, api_name)
+
+        for field_item_key in layout_item_dict.keys():
+            field = field_props.get(field_item_key)
+            field_layout_item = layout_item_dict.get(field_item_key)
+            adding_field_name = field.get("api_name")
+
+            self.logger.info(f"Adding {adding_field_name} to {api_name}")
+
+            behavior = "Edit"
+            required = process_bool_arg(field.get("required") or False)
+            read_only = process_bool_arg(field.get("read_only") or False)
+            if required and read_only is False:
+                behavior = "Required"
+            elif read_only and required is False:
+                behavior = "Readonly"
```

Comment:
`ReadOnly`? I'm not sure if this is case-sensitive.

Reply:
This is case sensitive, you can double check the reference in the [API docs](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#LayoutItem)